src/pymmcore_widgets/hcs/_util.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import annotations
+
+import math
+from itertools import product
+from typing import TYPE_CHECKING
+
+import numpy as np
+from qtpy.QtCore import QRectF, Qt
+from qtpy.QtWidgets import (
+    QGraphicsScene,
+    QGraphicsView,
+    QWidget,
+)
+from useq import WellPlate
+
+from ._graphics_items import _WellGraphicsItem
+
+if TYPE_CHECKING:
+    from qtpy.QtGui import QBrush, QPen, QResizeEvent
+    from useq._grid import GridPosition
+
+    from ._graphics_items import Well
+
+import useq
+from useq._plate import _KNOWN_PLATES
```

Comment:
don't import private stuff from external packages.  I added `useq.known_well_plate_keys()` for this

Reply:
basically... never ever import private stuff from external packages :joy: even if it's our stuff.  i know I say this a lot... but i mean it.  if you find yourself doing it, open an issue somewhere saying why you feel like you need to do it, and create a public export somewhere else